File: Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/model/invocation/MessageInvocation.java
package org.cqframework.cql.cql2elm.model.invocation;

import static java.util.Objects.requireNonNull;

import java.util.Arrays;
import java.util.List;
import org.hl7.elm.r1.Expression;
import org.hl7.elm.r1.Message;

public class MessageInvocation extends OperatorExpressionInvocation<Message> {
    public MessageInvocation(Message expression) {
        super(expression);
    }

    @Override
    public List<Expression> getOperands() {
        return Arrays.asList(
                expression.getSource(),
                expression.getCondition(),
                expression.getCode(),
                expression.getSeverity(),
                expression.getMessage());
    }

    @Override
    public void setOperands(List<Expression> operands) {
        requireNonNull(operands, "operands cannot be null.");
        require(operands.size() == 5, "Message operator requires five operands.");
        expression.setSource(operands.get(0));
        expression.setCondition(operands.get(1));
        expression.setCode(operands.get(2));
        expression.setSeverity(operands.get(3));
        expression.setMessage(operands.get(4));
    }
}

File: Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/model/invocation/MessageInvocation.kt
package org.cqframework.cql.cql2elm.model.invocation

import java.util.*
import org.hl7.elm.r1.Expression
import org.hl7.elm.r1.Message

class MessageInvocation(expression: Message) : OperatorExpressionInvocation<Message>(expression) {
    override var operands: List<Expression>
        get() =
            listOf(
                expression.source,
                expression.condition,
                expression.code,
                expression.severity,
                expression.message
            )
        set(operands) {
            require(operands.size == 5) { "Message operator requires five operands." }
            expression.source = operands[0]
            expression.condition = operands[1]
            expression.code = operands[2]
            expression.severity = operands[3]
            expression.message = operands[4]
        }
}
